File: PortfolioManager/PortfolioManager.Common/DTO/PortfolioDTO.cs
namespace PortfolioManager.Common.DTO;

public class PortfolioDTO
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;

    public double TotalEntryPrice {get; set; }
    public double TotalCurrentPrice {get; set; }
    public double TotalGain {get; set; } //TotalCurrentValue - TotalEntryValue
    public double TotalGainPercentage { get; set; }
    public string UserId { get; set; } = string.Empty;
    
    public ICollection<StockDTO>? Stocks {get; set;}
}

File: PortfolioManager/PortfolioManager.Client/portfolio-manager-client/src/DTO/PortfolioDto.ts
import StockDto from './StockDto';

export default interface PortfolioDto {
    id: number;
    name: string;
    totalEntryPrice: number;
    totalCurrentPrice: number;
    totalGain: number;
    totalGainPercentage: number;
    userId: string;
    stocks: StockDto[];
}
